scripts/test_osv_freeze.py

```
@@ -0,0 +1,74 @@
+    assert ergebnis["fastapi"] == "0.139.0"
+    assert ergebnis["nur-im-cu128-baum"] == "1.2.3"  # Union-Fallback: PyPI kennt es nicht
+
+
+def test_waechter_schlaegt_unter_100_an():
+    # Der Wächter unterscheidet "nichts aufgeloest" (Manifest-Laenge ~12) von
+    # "aufgeloest" (~127). Ohne ihn wäre ein leerer Report ein grüner Scan über
+    # nichts — der "pass heisst nicht geschaut"-Fehler aus den CodeRabbit-Limits.
+    try:
+        of.pruefe_anzahl(12)
+    except SystemExit as e:
+        assert e.code == 1
```

Ruff (0.16.1)

[warning] Found assertion on exception `e` in `except` block, use `pytest.raises()` instead

(PT017)